spyt-package/src/main/python/spyt/utils.py

```diff
@@ -218,8 +218,8 @@ def scala_buffer_to_list(buffer):
     return [buffer.apply(i) for i in range(buffer.length())]
 
 
-def default_user():
-    return os.getenv("YT_USER") or getpass.getuser()
+def default_user(client=None):
+    return os.getenv("YT_USER") or (get_user_name(client=client) if client else None) or getpass.getuser()
```

Comment:
There is a standard convention in almost all yt wrapper functions that they accept client as a last keyword argument, and client may be None, which means that the "default" client should be taken. In other words, I would expect that the "if client else None" part is not necessary;

Moreover, I would remove getpass.getuser() fallback, because there is no way that it can be  useful. If the YT client is not configured, nothing would work at all, so get_user_name must succeed, and getuser is redundant.